browser/src/control/Control.PartsPreview.js

```diff
@@ -289,12 +291,15 @@ L.Control.PartsPreview = L.Control.extend({
 			}
 			$trigger.contextMenu(true);
 			that._setPart(e);
+			$.contextMenu('destroy', '#' + img.id);
```

Comment:
why it is needed?

Reply:
Whenever a user right-clicks to open the context menu, multiple icon elements get appended to each menu item. To prevent this duplication, I added a call to `$.contextMenu('destroy', '#' + img.id)`  before generating the menu. It’s not the most elegant solution, but it effectively avoids the issue for now. I’ve opened this draft PR to get a second opinion on whether there’s a cleaner or more appropriate approach


https://github.com/user-attachments/assets/7a04374f-6594-486a-a667-0d149f37008d

Reply:
I would prefer to "NOT INSERT 2 times" than "ADD 2 times and remove 1" :)
Can we do it? What is the problem?

Reply:
the problem is in the Control.PartsPreview.js because in there (as opposed to what it happens on  Control.ContextMenu.js) the menu is never removed from the DOM) but it's also not added twice to the DOM probably because there is no remove context menu but only hide context menu. Therefore, we should probably handle the icon in the same way (only call your function when it is added to the DOM and not every time it's shown)